packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -130,49 +123,63 @@ function SelectionTool(props: Props) {
     'Escape',
     () => {
       startEvtRef.current = undefined;
-      setSelectionState({ selection: undefined, isCancelled: true });
+      setRawSelection(undefined);
     },
     [],
     { event: 'keydown' }
   );
 
-  useEffect(() => {
-    const { selection, isCancelled } = selectionState;
+  // Compute effective selection
+  const selection = useMemo(
+    () => rawSelection && transformSelectionRef.current(rawSelection),
+    [rawSelection, transformSelectionRef]
+  );
+
+  // Keep track of previous effective selection state
+  const prevSelection = usePrevious(selection);
```

Comment:
We now keep track of the previous "effective" selection. This makes more sense, since the effective selection is what the consumer cares most about. This is why I named the variables `selection` and `prevSelection`, and the local state `rawSelection`, instead of the opposite (i.e. `effectiveSelection` and `selection` for the local state).